core/src/main/java/com/linecorp/armeria/common/stream/StreamMessageUtil.java

```diff
@@ -53,6 +54,10 @@ static void closeOrAbort(Object obj, @Nullable Throwable cause) {
         }
 
         PooledObjects.close(obj);
+
+        if (obj instanceof SafeCloseable) {
+            ((SafeCloseable) obj).close();
+        }
```

Comment:
How about changing `PooledObjects.close(Object)` to close any `SafeCloseable`s rather than just `HttpData`?

Reply:
Looking at the `PooledObjects` docs - it's tailored for the pooled HttpData and supposed to be no-op for all other classes, I'm afraid that if we close any `SafeCloseable` there it'll cause confusion and result in bugs in the future.

What I can suggest as an alternative is to remove `PooledObjects.close(obj);` from here, it's effectively replaced by this new condition. What do you think?

Reply:
SGTM.